examples/car_fsm.erl

```diff
@@ -0,0 +1,543 @@
+%%% -*- coding: utf-8 -*-
+%%% -*- erlang-indent-level: 2 -*-
+%%% -------------------------------------------------------------------
+%%% Copyright 2020-     Manolis Papadakis <manopapad@gmail.com>,
+%%%                     Eirini Arvaniti <eirinibob@gmail.com>
+%%%                 and Kostis Sagonas <kostis@cs.ntua.gr>
+%%%
+%%% This file is part of PropEr.
+%%%
+%%% PropEr is free software: you can redistribute it and/or modify
+%%% it under the terms of the GNU General Public License as published by
+%%% the Free Software Foundation, either version 3 of the License, or
+%%% (at your option) any later version.
+%%%
+%%% PropEr is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%%% GNU General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License
+%%% along with PropEr.  If not, see <http://www.gnu.org/licenses/>.
+
+%%% @copyright 2020 Spiros Dontas and Kostis Sagonas
+%%% @version {@version}
+%%% @author Spiros Dontas
+
+-module(car_fsm).
+-behaviour(gen_statem).
+-behaviour(proper_fsm).
+
+-include_lib("proper/include/proper.hrl").
+
+
+%% -----------------------------------------------------------------------------
+%% Exports
+%% -----------------------------------------------------------------------------
+
+
+%% API
+-export([start_link/0, stop/0, accelerate/1, brake/1, travel/1, refuel/1]).
+%% gen_statem callbacks
+-export([init/1, callback_mode/0, stopped/3, slow/3, normal/3, fast/3,
+         too_fast/3, terminate/3, code_change/4]).
+%% proper_fsm callbacks
+-export([initial_state/0, initial_state_data/0, next_state_data/5,
+         precondition/4, postcondition/5, weight/3]).
+-export([stopped/1, slow/1, normal/1, fast/1, too_fast/1]).
+
+
+%% -----------------------------------------------------------------------------
+%% Definitions
+%% -----------------------------------------------------------------------------
+
+
+-define(DISTANCE, 1000).
+-define(CONSUMPTION, 10).
+-define(HOUR, 3600).
+-define(ACCELERATION, 5).
+-define(DECELERATION, 20).
+-define(MAX_FUEL, 70).
+-define(MAX_SPEED, 200).
+-define(AVG(X), lists:sum(X) / length(X)).
+-define(SLOW_THRESHOLD, 50).
+-define(NORMAL_THRESHOLD, 100).
+-define(FAST_THRESHOLD, 150).
+-define(CALL(C, A), {call, ?MODULE, C, A}).
```

Comment:
The PropEr(TM) use of this `?CALL` macro is unclear.  The module uses it at some places, but not in other cases that it would look reasonable to also use it (e.g. in `weight`, `precondition`, `postcondition`, etc.) it does not.  Either use it consistently, or take it out.